### DISCUSSION TITLE
Radio menu with submenus

### DISCUSSION THREAD

**Discussion by marcospassos** [ariakit/ariakit#4033] [Questions] [answered]
We're looking to set up a selector for a code block, like the one in the screenshot below:

![Expected](https://github.com/user-attachments/assets/e064b98d-b765-4360-846f-a45cf2eb0429)

 I couldn't find any examples using nested radio buttons, but we ran into two issues with the straightforward implementation:

- The submenu doesn't open by default when the selected option is inside it.
- Selecting a submenu doesn't close the parent menu (a workaround would be to just call the parent menu state, but that seems kind of hacky). 

https://github.com/user-attachments/assets/e9c60bd8-17b0-4fe7-a9d0-2a7c940b1a42

Am I missing something here? Any advice on how to implement this use case?

**Comment by benrodrs** [comment]
Could you provide a StackBlitz with the code you're implementing?

**Comment by georgekaran** [comment]
Here's the snippet in StackBlitz: https://stackblitz.com/edit/hduypu?file=menu-nested/style.css,menu-nested/menu.tsx,menu-nested/index.tsx,node_modules/@ariakit/react-core/esm/menu/menu-store.d.ts&theme=dark

I was able to close the menus by doing the following:
```tsx
onClick={() => {
  store?.hide();
  store?.parent?.hide();
}}
```

**Reply by marcospassos** [reply]
> I was able to close the menus by doing the following:

This works for a two-level menu but not in a recursive fashion

**Reply by benrodrs** [reply]
Thank you for the code.

> * Selecting a submenu doesn't close the parent menu (a workaround would be to just call the parent menu state, but that seems kind of hacky).

You can enable the [`hideOnClick`](https://ariakit.org/reference/menu-item-radio#hideonclick) prop to close all menus.

> * The submenu doesn't open by default when the selected option is inside it.

You can create custom logic for that, but I believe that's not how menus are supposed to work. Especially when you mention `selected option`, it suggests using [Select](https://ariakit.org/components/select) instead of Menu. See [Should I use Menu or Select?](https://ariakit.org/components/menu#should-i-use-menu-or-select). I also recommend [SelectGroup](https://ariakit.org/examples/select-group) instead of a submenu.

**Reply by diegohaz** [reply, marked as answer]
Yeah, this "Framework" select is definitely a select widget, not a menu. On Ariakit.org, we have two dropdown buttons in the code block toolbar. One is a select widget where the button content reflects the selected option, and the selected option is automatically focused when the dropdown opens, and the other is a menu button (a set of actions or settings):

<img height="180" alt="screenshot of the code block toolbar with the language select dropdown open" src="https://github.com/user-attachments/assets/070aeb21-173d-40a6-aa21-2e29decf4cc4">
<img height="180" alt="screenshot of the code block toolbar with the edit example menu button open" src="https://github.com/user-attachments/assets/daa09ef5-a369-4e21-8cd8-c065e3a60cb6">
